### PR TITLE
[Compatibility] Download Real Media Library folder

### LINKED ISSUE #9: [Compatibility] Download Real Media Library folder
Hi there!

I am the developer of the WordPress Real Media Library Plugin (https://codecanyon.net/item/wordpress-real-media-library-media-categories-folders/13155134). I got often asked for a feature to download a folder as zip file. Instead of reinventing the wheel I am a fan of Extensions and Add-Ons. I see your plugin is doing the download job weel.

Before making a PR to your plugin I want to ask you what do you think about that? The compatibility will be very lightweight.

Best regards,
Matthew :smile:

### PR DESCRIPTION
**Pull request by matzeeable**
The following implementation allows to download a Real Media Library folder:

- Without RML subfolders: flat, physical structure
- With RML subfolders: flat, physical structure, RML structure

Resolves #9 

If you want you can contact me support ( a t ) matthias-web.com so I can send you the latest Real Media Library plugin version and you can test.

EDIT: I have added a screenshot - you can remove it if you do not want to make it visible on wp.org. Also the version number is not updated.

Best regards,
Matthew :-)

### CODE REVIEW

File: lib/MassEdge/WordPress/Plugin/ExportMediaLibrary/Module/AdminPageExport.php [hunk 1]
```diff
@@ -116,12 +116,12 @@ function export($exportName, $folderStructure, $compress) {
         ]);
 
         $query = new \WP_Query();
-        $attachmentIds = $query->query([
+        $attachmentIds = $query->query(array_merge([
```
Comment (andrejpavlovic) [inline lib/MassEdge/WordPress/Plugin/ExportMediaLibrary/Module/AdminPageExport.php]: Hook into here.

File: lib/MassEdge/WordPress/Plugin/ExportMediaLibrary/Module/AdminPageExport.php [hunk 2]
```diff
@@ -144,6 +144,7 @@ function export($exportName, $folderStructure, $compress) {
                             $file = $attachmentPath;
                         }
                     }
+                    $file = apply_filters('massedge/wp/eml/export/nested/file', $file, $attachmentId, $attachmentIds, $attachmentPath);
```
Comment (andrejpavlovic) [inline lib/MassEdge/WordPress/Plugin/ExportMediaLibrary/Module/AdminPageExport.php]: and hook into here

File: lib/MassEdge/WordPress/Plugin/ExportMediaLibrary/Module/AdminPageExport.php [hunk 3]
```diff
@@ -105,7 +105,7 @@ function page() {
         echo ob_get_clean();
     }
 
-    function export($exportName, $folderStructure, $compress) {
+    function export($exportName, $folderStructure, $compress, $overrides = []) {
```
Comment (andrejpavlovic) [inline lib/MassEdge/WordPress/Plugin/ExportMediaLibrary/Module/AdminPageExport.php]: This function needs to be exposed at the plugin level, so other plugins can access it more easily.

**Review by andrejpavlovic — COMMENTED** [review]
From what I can see, you only need to hook into two parts of the code, and the `export` function needs to be more easily accessible by other plugins. All of the other RML-specific code should be in your plugin, not in this one.

**Comment by andrejpavlovic** [comment]
I'm putting together a version that will allow you to hook into more easily. Should have it done later today.

**Comment by matzeeable** [comment]
Hi! Ok, I will put this into RML coding. Please let me know when you have finished the coding for that. :-)

**Comment by matzeeable** [comment]
Hi @andrejpavlovic !

I know this is a closed issue but I write here because I do not want to loose the context of the next question. We have now completely rebranded our plugin https://devowl.io/wordpress-real-media-library/ and also published a lite version: https://wordpress.org/plugins/real-media-library-lite/

It would be nice if you could mention the lite version in the description of your plugin that there is an extended compatibility. What do you think? 😃

**Pull request closed without merging**